### PR TITLE
Add player and weapon color funcs to shared

### PR DESCRIPTION
**Pull request by banoargeek**
Starfall currently lacks player methods for getting the Player and Weapon colors the same way E2 does. It just uses built-in Garry's Mod methods.

### CODE REVIEW

**Comment by samuelWilliams99** [comment]
hey @plally can you review this please, it'd make me sooooo happy

**Review by plally — APPROVED** [review]

**Comment by Syvil** [comment]
I think colors are one of the things you are meant to wrap when returning.
instance.Types.Color.Wrap( Color ) is the function for that.

**Review by plally — CHANGES_REQUESTED** [review]
good catch @Syvil, should be looked into before merging

**Comment by banoargeek** [comment]
@plally It says you requested a change but I cannot seem to see it.

**Comment by banoargeek** [comment]
Oh, my bad. Let me read up on the comment thread. :)

**Comment by wrefgtzweve** [comment]
> @plally It says you requested a change but I cannot seem to see it.

image

**Comment by banoargeek** [comment]
Thank you @Syvil, I'll rectify that

**Comment by banoargeek** [comment]
Ty @Redox

**Comment by banoargeek** [comment]
Apologies for the duplicate commit, lmao

**Comment by banoargeek** [comment]
Okay, I messed up in a way that I have not yet encountered. I'll fix it with a commit.

**Comment by banoargeek** [comment]
Please let me know if my local variables are defined satisfactorily.

**Comment by Syvil** [comment]
If you are going to localize instance.Types, you could instead localize instance.Types.Color.Wrap, which would remove an extra two table lookups from each function.
`colwrap( ply:GetWeaponColor() )` instead of `types.Color.Wrap( ply:GetWeaponColor() )`

File: lua/starfall/libs_sh/ply.lua [hunk 1]
```diff
@@ -60,6 +62,24 @@ function player_methods:getAmmo()
     return ply:GetAmmo()
 end
 
+--- Gets the vector of a player's color.
+-- @return The vector of a player's color
+function player_methods:getPlayerColor()
+    checktype( self, ply_meta )
+    local ply = getPly( self )
+    local color = types.Color.Wrap( ply:GetPlayerColor() )
+    return
```
Comment (plally) [inline lua/starfall/libs_sh/ply.lua:71]: you aren't returning anything here

**Review by plally — CHANGES_REQUESTED** [review]

**Comment by plally** [comment]
I agree with syvil, I think localizing color would be good. `local colorWrap = types.Color.Wrap`
You could also take just do `return colorWrap( color )` instead  of using a local variable

**Comment by banoargeek** [comment]
Agh, I must be having trouble paying attention on account of getting my covid shot, let me re-readdress that.

**Comment by banoargeek** [comment]
That should address the current concerns.

**Review by plally — APPROVED** [review]

**Pull request merged**